gcslock.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+#
+# Copyright 2016 Google Inc.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file illustrates how to the use the gcslock global mutex
+# via the gsutil command in a shell script. It depends on installation
+# of the Google Cloud SDK (https://cloud.google.com/sdk/), which 
+# includes the gsutil command.
+
+
+# The lock function expects the first (and only) argument to be
+# the name of a bucket writable by the user running this script.
+# It creates the lock object in the passed bucket with a special
+# header to obtain mutual exclusion.
+lock() {
+  if [ "$1" = "" ]
+  then
+    echo missing bucket argument
```

Comment:
I was thinking, if I had a script using this, I might not know where this message comes from. Maybe add a "lock: " prefix? So that it's more clear where the error is, "lock: missing bucket argument".